addons/purchase_stock/models/product.py

```diff
@@ -107,7 +107,6 @@ def _compute_quantities_dict(self, lot_id, owner_id, package_id, from_date=False
     @api.depends_context('suggest_based_on', 'warehouse_id')
     def _compute_monthly_demand(self):
         based_on = self.env.context.get("suggest_based_on", "30_days")
-        warehouse_id = self.env.context.get('warehouse_id')
```

Comment:
Adapt a bit the commit message to follow the [guidelines](https://www.odoo.com/documentation/master/contributing/development/git_guidelines.html#commit-message-structure)
`[IMP] purchase_stock: extend monthly demand domain`